### PR TITLE
Refactoring the provider to unit test create cluster

### PR DESCRIPTION
**Pull request by Skarlso**
### Description

Ground work for https://github.com/weaveworks/eksctl/issues/5015.

#### What happens in this PR?

1. Renamed `Provider` to `AWSProvider` since it provides AWS functionality
2. ClusterProvider is no longer a KubeProvider. It has been extracted into a separate part so it can be mocked more easily
3. doClusterCreate now gets the ctl passed in from the outside so we can mock it
4. Several small refactors and test changes for mock calls

Integration test run https://github.com/weaveworks/eksctl-ci/actions/runs/2330469236

_edit_: New test run https://github.com/weaveworks/eksctl-ci/actions/runs/2332032290

### Checklist
- [x] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/ctl/create/cluster.go
Comment (Skarlso) [inline pkg/ctl/create/cluster.go:138]: The above logger initialisation breaks now that newCtl happens before this. NewCtl contains a `once` for the logger. I'm working on the fix.

Comment (Skarlso) [reply pkg/ctl/create/cluster.go:138]: Fixed.

**Comment by Skarlso** [comment]
Integration tests passed except for a single test which timed out on TLS. That appears to be a fluke from AWS though.

File: pkg/eks/api.go
Comment (cPu1) [inline pkg/eks/api.go:51]: If we embed this field, we wouldn't have to reference this field via `ctl` everywhere and it would also make the diff smaller. 
```suggestion
	KubeProvider
```

**Review by cPu1 — APPROVED** [review]
Just one comment about considering embedding `KubeProvider`, but otherwise LGTM.

My [deadline-based Context PR](https://github.com/weaveworks/eksctl/pull/5282) will result in conflicts, so I'll wait for this to be merged before merging mine.

File: pkg/eks/api.go
Comment (Skarlso) [reply pkg/eks/api.go:51]: Hmmm, not a bad idea. I guess it would result in a smaller diff since I don't have to call it explicitly everywhere. 🤔

Comment (Skarlso) [reply pkg/eks/api.go]: Hm, not that much :) The AWSProvider rename touched more. :) But I've done it anyways.

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**
